src/ripple/shamap/SHAMap.h

```diff
@@ -80,6 +80,7 @@ class SHAMap
 {
 private:
     using Family = shamap::Family;
+    using NodeStack = std::stack<std::pair<SHAMapAbstractNode*, SHAMapNodeID>>;
```

Comment:
I'm a little surprised that you're using the `deque` implementation of the `stack`.  Intentional?  If so I'd be curious to know the reason for the choice.